sklearnex/linear_model/ridge.py

```diff
@@ -14,11 +14,416 @@
+                    normalize=normalize,
+                    copy_X=copy_X,
+                    max_iter=max_iter,
+                    tol=tol,
+                    solver=solver,
+                    random_state=random_state,
+                )
+
+        def fit(self, X, y, sample_weight=None):
+            if sklearn_check_version("1.2"):
+                self._validate_params()
+
+            # It is necessary to properly update coefs for predict if we
+            # fallback to sklearn in dispatch
+            if hasattr(self, "_onedal_estimator"):
+                del self._onedal_estimator
+
+            dispatch(
+                self,
+                "fit",
+                {
+                    "onedal": self.__class__._onedal_fit,
+                    "sklearn": _sklearn_Ridge.fit,
+                },
+                X,
+                y,
+                sample_weight,
+            )
+            return self
+
+        @wrap_output_data
+        def predict(self, X):
+            check_is_fitted(self)
+
+            return dispatch(
+                self,
+                "predict",
+                {
+                    "onedal": self.__class__._onedal_predict,
+                    "sklearn": _sklearn_Ridge.predict,
+                },
+                X,
+            )
+
+        @wrap_output_data
+        def score(self, X, y, sample_weight=None):
+            check_is_fitted(self)
+
+            return dispatch(
+                self,
+                "score",
+                {
+                    "onedal": self.__class__._onedal_score,
+                    "sklearn": _sklearn_Ridge.score,
+                },
+                X,
+                y,
+                sample_weight=sample_weight,
+            )
+
+        def _onedal_fit_supported(self, patching_status, method_name, *data):
+            assert method_name == "fit"
+            assert len(data) == 3
+            X, y, sample_weight = data
+
+            normalize_is_set = (
+                hasattr(self, "normalize")
+                and self.normalize
+                and self.normalize != "deprecated"
+            )
+            positive_is_set = hasattr(self, "positive") and self.positive
+
+            n_samples = _num_samples(X)
+            n_features = _num_features(X, fallback_1d=True)
+
+            # Check if equations are well defined
+            is_underdetermined = n_samples < (n_features + int(self.fit_intercept))
+
+            patching_status.and_conditions(
+                [
+                    (
+                        self.solver == "auto",
+                        f"'{self.solver}' solver is not supported. "
+                        "Only 'auto' solver is supported.",
+                    ),
+                    (
+                        not issparse(X) and not issparse(y),
+                        "Sparse input is not supported.",
+                    ),
+                    (
+                        not is_underdetermined,

```

Comment:
This is worth checking against https://github.com/intel/scikit-learn-intelex/pull/2083 where underdetermined checks are now only necessary for GPU.